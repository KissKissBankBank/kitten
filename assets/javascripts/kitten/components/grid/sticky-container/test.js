import React from 'react'
import renderer from 'react-test-renderer'
import { StickyContainer } from './index'

describe('<StickyContainer />', () => {
  let component

  describe('by default', () => {
    beforeEach(() => {
      component = renderer.create(<StickyContainer />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with some props', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <StickyContainer
            className="custom__class"
            style={{ color: 'red' }}
            aria-hidden
            isStickyOnScroll="up"
            top={66}
          />,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('already sticky', () => {
    beforeEach(() => {
      component = renderer
        .create(<StickyContainer isSticky={true} bottom={0} />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
