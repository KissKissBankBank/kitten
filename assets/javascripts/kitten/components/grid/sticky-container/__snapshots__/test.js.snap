// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StickyContainer /> already sticky matches with snapshot 1`] = `
<div
  className="sticky-container__StyledStickyContainer-trwgvt-0 leOXaT"
/>
`;

exports[`<StickyContainer /> by default matches with snapshot 1`] = `
<div
  className="sticky-container__StyledStickyContainer-trwgvt-0 irbHjO"
/>
`;

exports[`<StickyContainer /> with some props matches with snapshot 1`] = `
<div
  className="sticky-container__StyledStickyContainer-trwgvt-0 irbHjO"
/>
`;
